Sources/Internal/VisibleItemsProvider.swift

```diff
@@ -918,6 +937,80 @@ final class VisibleItemsProvider {
+      guard let framesForDays = framesForDaysForVisibleMonths[month] else { continue }
+
+      // We need to expand the frame of the month so that we have enough room at the edges to draw
+      // the background without getting clipped.
+      let extraWidth: CGFloat
+      let extraHeight: CGFloat
+      if content.monthsLayout.isHorizontal {
+        extraWidth = content.interMonthSpacing // half before leading edge, half after trailing edge
+        extraHeight = size.height - monthFrame.height
+      } else {
+        extraWidth = size.width - monthFrame.width
+        extraHeight = content.interMonthSpacing // half before top edge, half after bottom edge
+      }
+
+      let expandedMonthFrame = CGRect(
+        x: monthFrame.minX - (extraWidth / 2),
+        y: monthFrame.minY - (extraHeight / 2),
+        width: monthFrame.width + extraWidth,
+        height: monthFrame.height + extraHeight)
+      let frameToBoundsTransform = CGAffineTransform(
+        translationX: -expandedMonthFrame.minX,
+        y: -expandedMonthFrame.minY)
+
+      // Get the month header frame
+      let monthHeaderFrame = frameProvider.frameOfMonthHeader(inMonthWithOrigin: monthFrame.origin)
+      let finalMonthHeaderFrame = monthHeaderFrame
+        .applying(frameToBoundsTransform)
+        .alignedToPixels(forScreenWithScale: scale)
+
+      // Get the days-of-the-week item frames
+      var dayOfWeekPositionsAndFrames = [(dayOfWeekPosition: DayOfWeekPosition, frame: CGRect)]()
+      for dayOfWeekPosition in DayOfWeekPosition.allCases {
+        let dayOfWeekFrame = frameProvider.frameOfDayOfWeek(
+          at: dayOfWeekPosition,
+          inMonthWithOrigin: monthFrame.origin)
+        let finalDayOfWeekFrame = dayOfWeekFrame
+          .applying(frameToBoundsTransform)
+          .alignedToPixels(forScreenWithScale: scale)
+        dayOfWeekPositionsAndFrames.append((dayOfWeekPosition, finalDayOfWeekFrame))
+      }
+
+      // Get all frames for days in the month
+      var daysAndFrames = [(day: Day, frame: CGRect)]()
+      for (day, dayFrame) in framesForDays {
+        let finalDayFrame = dayFrame
+          .applying(frameToBoundsTransform)
+          .alignedToPixels(forScreenWithScale: scale)
+        daysAndFrames.append((day, finalDayFrame))
+      }
+
+      let monthLayoutContext = CalendarViewContent.MonthLayoutContext(
+        month: month,
+        monthHeaderFrame: finalMonthHeaderFrame,
+        dayOfWeekPositionsAndFrames: dayOfWeekPositionsAndFrames,
+        daysAndFrames: daysAndFrames.sorted(by: { $0.day < $1.day }),
+        bounds: CGRect(origin: .zero, size: expandedMonthFrame.size))
+      if let itemModel = monthBackgroundItemProvider(monthLayoutContext) {
+        let visibleItem = VisibleItem(
+          calendarItemModel: itemModel,
+          itemType: .monthBackground(month),
+          frame: expandedMonthFrame)
+        visibleItems.insert(visibleItem)
+      }
+    }
+  }
```

Comment:
this is where we build the `MonthLayoutContext`, which has information about the location of all subviews in the month.